pkg/operations/handlers.go

```diff
@@ -86,68 +86,53 @@ func filterAndGroupBlocks(index *bucketindex.Index, query *blockQuery) *blockLis
-					MinTimeAge:       humanize.RelTime(blk.MinTime.Time(), time.Now(), "ago", ""),
+					MinTime:                 truncatedMinTime,
+					FormattedMinTime:        truncatedMinTime.Format(time.RFC3339),
+					Blocks:                  make([]*blockDetails, 0),
+					MinTimeAge:              humanize.RelTime(blk.MinTime.Time(), time.Now(), "ago", ""),
+					MaxBlockDurationMinutes: int(math.Round(blk.MaxTime.Sub(blk.MinTime).Minutes())),
 				}
 				blockGroups = append(blockGroups, blkGroup)
 			}
-			blkGroup.Blocks = append(blkGroup.Blocks, &blockDetails{
+			blockDetails := &blockDetails{
 				ID:               blk.ID.String(),
-				MinTime:          formattedMinTime,
+				MinTime:          minTime.Format(time.RFC3339),
 				MaxTime:          blk.MaxTime.Time().UTC().Format(time.RFC3339),
 				Duration:         int(math.Round(blk.MaxTime.Sub(blk.MinTime).Minutes())),
 				UploadedAt:       blk.GetUploadedAt().UTC().Format(time.RFC3339),
 				CompactionLevel:  blk.CompactionLevel,
 				CompactorShardID: blk.CompactorShardID,
-			})
-			blockGroupMap[formattedMinTime] = blkGroup
+			}
+			blkGroup.Blocks = append(blkGroup.Blocks, blockDetails)
+			if blockDetails.Duration > blkGroup.MaxBlockDurationMinutes {
+				blkGroup.MaxBlockDurationMinutes = blockDetails.Duration
+			}
+			blockGroupMap[truncatedMinTime] = blkGroup
 		}
 	}
 
 	sortBlockGroupsByMinTimeDec(blockGroups)
 
-	return postProcessBlockGroups(blockGroups)
-}
-
-func postProcessBlockGroups(blockGroups []*blockGroup) *blockListResult {
 	maxBlocksPerGroup := 0
-	for i := 0; i < len(blockGroups); i += 1 {
-		blockGroup := blockGroups[i]
-		if i < len(blockGroups)-1 && !strings.Contains(blockGroup.FormattedMinTime, "0:00Z") {
-			nextGroup := blockGroups[i+1]
-			nextGroup.Blocks = append(nextGroup.Blocks, blockGroup.Blocks...)
-			blockGroup.Blocks = make([]*blockDetails, 0)
-		}
-
+	maxBlockGroupDuration := 0
+	for _, blockGroup := range blockGroups {
 		sortBlockDetailsByMinTimeDec(blockGroup.Blocks)
-
 		if len(blockGroup.Blocks) > maxBlocksPerGroup {
 			maxBlocksPerGroup = len(blockGroup.Blocks)
 		}
-	}
-
-	finalBlockGroups := make([]*blockGroup, 0)
-	for _, blockGroup := range blockGroups {
-		if len(blockGroup.Blocks) > 0 {
-			finalBlockGroups = append(finalBlockGroups, blockGroup)
+		if blockGroup.MaxBlockDurationMinutes > maxBlockGroupDuration {
+			maxBlockGroupDuration = blockGroup.MaxBlockDurationMinutes
 		}
 	}
-	groupDuration := 60
-	if len(finalBlockGroups) > 1 {
-		groupOne := finalBlockGroups[len(finalBlockGroups)-2]
-		groupTwo := finalBlockGroups[len(finalBlockGroups)-1]
-		groupDuration = int(math.Round(groupOne.MinTime.Sub(groupTwo.MinTime).Minutes()))
-	}
+
 	return &blockListResult{
-		BlockGroups:       finalBlockGroups,
-		MaxBlocksPerGroup: maxBlocksPerGroup,
-		GroupDuration:     groupDuration,
+		BlockGroups:          blockGroups,
+		MaxBlocksPerGroup:    maxBlocksPerGroup,
+		GroupDurationMinutes: maxBlockGroupDuration,
```

Comment:
Nit: Maybe we can make that a time.Duration:

```suggestion
		GroupDurationMinutes: time.Minutes*maxBlockGroupDuration,
```

Currently it shows as `5` or `60`, I would prefer `5m` `1h`

Reply:
We need the actual numbers for the grid view (we do some calculations there). I can add a human readable duration in addition to this though for the table view.

Reply:
Wasn't realizing, thanks for fixing it